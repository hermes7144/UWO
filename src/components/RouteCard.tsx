import React from 'react';
import useRoute from '../Hooks/useRoute';
import { useInfoContext } from './context/InfoContext';
import { BsFillTrashFill } from 'react-icons/bs';
import { useNavigate } from 'react-router-dom';

type RouteType = {
  route: {
    id: string;
    title: string;
    remark: string;
    citys: number[];
  };
};

export default function RouteCard({ route }: RouteType) {
  const navigate = useNavigate();

  const { setCitys, setRoute } = useInfoContext();
  const { removeItem } = useRoute();

  const handleRoute = () => {
    navigate(`/routes/${route.id}`, { state: { route } });
    setCitys(route.citys);
    setRoute(route);
  };

  const handleDelete = () => {
    const id = route.id;
    removeItem.mutate(
      { id },
      {
        onSuccess: () => {
          setRoute({ id: '', title: '', remark: '' });
          setCitys([]);
        },
      }
    );
  };

  return (
    <li className={'round-lg shadow-md overflow-hidden cursor-pointer p-2 m-1'} onClick={handleRoute}>
      <div className='mt-2 px-2 flex flex-row items-center justify-between'>
        <h3>{route.title}</h3>
        <BsFillTrashFill className='cursor-pointer opacity-50 hover:opacity-100' onClick={handleDelete} />
      </div>
    </li>
  );
}
