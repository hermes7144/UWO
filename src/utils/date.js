// import { format, register } from 'timeago.js';
// import koLocale from 'timeago.js/lib/lang/ko';

// register('ko', koLocale);

// export function formatAgo(date, lang = 'en_US') {
//   return format(date, lang);
// }

export function formatDate(rawDate) {
  const date = new Date(rawDate);
  const year = date.getFullYear().toString().slice(-2); // 연도에서 뒤의 두 자리만 추출
  const month = ('0' + (date.getMonth() + 1)).slice(-2); // 월을 두 자리 숫자로 변환
  const day = ('0' + date.getDate()).slice(-2); // 일을 두 자리 숫자로 변환
  return `${year}/${month}/${day}`;
}