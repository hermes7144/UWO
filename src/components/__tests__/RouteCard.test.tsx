import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Route, useLocation } from 'react-router-dom';
import { fakeRoute as route } from '../../tests/fakeRoutes';
import { withAllContexts, withRouter } from '../../tests/utils';
import RouteCard from '../RouteCard';

describe('RouteCard', () => {
  const fakeUseCity = jest.fn();

  it('renders correctly', () => {
    fakeUseCity.mockImplementation(() => {
      return { routeQuery: { data: route } };
    });
    const { asFragment } = renderRouteCard();

    expect(asFragment()).toMatchSnapshot();
  });

  it('navigates to detailed route page with route state when clicked', async () => {
    render(
      withAllContexts(
        withRouter(
          <>
            <Route path='/' element={<RouteCard route={route} />} />
            <Route path={`/routes/${route.id}`} element={<p>{route.description}</p>} />
          </>
        ),
        fakeUseCity
      )
    );

    const card = screen.getByRole('listitem');
    await userEvent.click(card);

    expect(screen.getByText(route.description)).toBeInTheDocument();
  });

  function renderRouteCard() {
    return render(withAllContexts(withRouter(<Route path='/' element={<RouteCard route={route} />} />), fakeUseCity));
  }
});
