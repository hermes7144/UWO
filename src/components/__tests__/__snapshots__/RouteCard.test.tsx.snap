// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteCard renders correctly 1`] = `
<DocumentFragment>
  <li
    class="round-lg shadow-lg overflow-hidden cursor-pointer hover:shadow-sm"
  >
    <div
      class="px-1 text-xl overflow-hidden overflow-ellipsis whitespace-nowrap flex justify-between mx-2"
    >
      <div>
        <span />
        <span />
        <span
          class="pl-3"
        />
        ~
        <span />
      </div>
      <span>
        title
      </span>
      <span>
        23/04/06
      </span>
    </div>
  </li>
</DocumentFragment>
`;
